tests/token/erc1155/test_ERC1155_Mintable_Burnable.py

```diff
@@ -0,0 +1,1555 @@
+    account2 = cached_contract(_state, account_def, account2)
+    erc1155 = cached_contract(_state, erc1155_def, erc1155)
+    receiver = cached_contract(_state, receiver_def, receiver)
+    return erc1155, account1, account2, receiver
+
+
+@pytest.fixture(scope='module')
+async def erc1155_minted_init(contract_defs, erc1155_init):
+    account_def, erc1155_def, receiver_def = contract_defs
+    state, owner, account, erc1155, receiver = erc1155_init
+    _state = state.copy()
+    owner = cached_contract(_state, account_def, owner)
+    account = cached_contract(_state, account_def, account)
+    erc1155 = cached_contract(_state, erc1155_def, erc1155)
+    receiver = cached_contract(_state, receiver_def, receiver)
+    await signer.send_transaction(
+        owner, erc1155.contract_address, 'mintBatch',
+        [
+            account.contract_address,  # to
+            *uarr2cd(TOKEN_IDS),  # ids
+            *uarr2cd(MINT_AMOUNTS),  # amounts
+            0  # data
+        ]
+    )
+    return _state, erc1155, owner, account, receiver
+
+
+@pytest.fixture
+def erc1155_minted_factory(contract_defs, erc1155_minted_init):
+    account_def, erc1155_def, receiver_def = contract_defs
+    state, erc1155, owner, account, receiver = erc1155_minted_init
+    _state = state.copy()
+    owner = cached_contract(_state, account_def, owner)
+    account = cached_contract(_state, account_def, account)
+    erc1155 = cached_contract(_state, erc1155_def, erc1155)
+    receiver = cached_contract(_state, receiver_def, receiver)
+    return erc1155, owner, account, receiver
+
+
+# Tests
+
+#
+# Constructor
+#
+
+@pytest.mark.asyncio
+async def test_constructor(erc1155_factory):
+    erc1155, _, _, _ = erc1155_factory
+
+    execution_info = await erc1155.uri().invoke()
+    assert execution_info.result.uri == 0
+
+#
+# ERC165
+#
+
+
+@pytest.mark.asyncio
+async def test_supports_interface(erc1155_factory):
+    erc1155, _, _, _ = erc1155_factory
+
+    for supported_id in SUPPORTED_INTERFACES:
+        execution_info = await erc1155.supportsInterface(
+            supported_id
+        ).invoke()
+        assert execution_info.result.is_supported == TRUE
+
+    for unsupported_id in UNSUPPORTED_INTERFACES:
+        execution_info = await erc1155.supportsInterface(
+            unsupported_id
+        ).invoke()
+        assert execution_info.result.is_supported == FALSE
+
+#
+# Set/Get approval
+#
+
+
+@pytest.mark.asyncio
+async def test_set_approval_for_all(erc1155_factory):
+    erc1155, account, _, _ = erc1155_factory
+
+    operator = ACCOUNT
+    approval = TRUE
+
+    await signer.send_transaction(
+        account, erc1155.contract_address, 'setApprovalForAll',
+        [operator, approval]
+    )
+
+    execution_info = await erc1155.isApprovedForAll(
+        account.contract_address,
+        operator
+    ).invoke()
+
+    assert execution_info.result.is_approved == approval
+
+    operator = ACCOUNT
+    approval = FALSE
```

Comment:
My goal there was to aim for consistency. It's actually unclear to me whether these local variables add clarity at all, or whether the names of constants are sufficient. I can see an argument that it makes the role of the arguments clearer (e.g. `operator` instead of `account.contract_address # operator`). Thoughts?